src/olympia/git/management/commands/git_extraction.py

```diff
@@ -0,0 +1,109 @@
+import waffle
+
+from celery import chain
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+import olympia.core.logger
+
+from olympia.amo.decorators import use_primary_db
+from olympia.amo.utils import chunked
+from olympia.files.utils import lock
+from olympia.git.models import GitExtractionEntry
+from olympia.git.tasks import (
+    extract_versions_to_git,
+    on_extraction_error,
+    remove_git_extraction_entry,
+)
+
+
+log = olympia.core.logger.getLogger('z.git.git_extraction')
+
+BATCH_SIZE = 10  # Number of versions to extract in a single task.
+LOCK_NAME = 'git-extraction'  # Name of the lock() used.
+SWITCH_NAME = 'enable-git-extraction-cron'  # Name of the waffle switch.
+
+
+class Command(BaseCommand):
+    help = 'Extract add-on versions into Git repositories'
+
+    @use_primary_db
+    def handle(self, *args, **options):
+        if not waffle.switch_is_active(SWITCH_NAME):
+            log.info(
+                'Not running git_extraction command because switch "{}" is '
+                'not active.'.format(SWITCH_NAME)
+            )
+            return
+
+        # Get a lock before doing anything, we don't want to have multiple
+        # instances of the command running in parallel.
+        with lock(settings.TMP_PATH, LOCK_NAME) as lock_attained:
+            if not lock_attained:
+                # We didn't get the lock...
+                log.error('{} lock present, aborting.'.format(LOCK_NAME))
+                return
+
+            # If an add-on ID is present more than once, the `extract_addon()`
+            # method will skip all but the first occurrence because the add-on
+            # will be locked for git extraction.
+            entries = GitExtractionEntry.objects.order_by('created').all()
+            for entry in entries:
+                self.extract_addon(entry)
+
+    def extract_addon(self, entry):
+        """
+        This method takes a GitExtractionEntry object and creates a chain of
+        Celery tasks to extract each version in a git repository that haven't
+        been extracted yet (including the deleted versions).
+
+        It does not run if the add-on is locked for git extraction.
+        """
+        addon = entry.addon
+        log.info('Starting extraction of add-on "{}".'.format(addon.pk))
+
+        if addon.git_extraction_is_in_progress:
```

Comment:
I thought we could remove this and use `entry.in_progress` but that didn't work (and I added a test case for that). We could use the model directly here but I find the call to this property more readable (and we could use the property elsewhere).

Reply:
Add a comment explaining why we can't use `entry.in_progress` here :)